Fix minor issues in OpenGL buffer creation

MrLever opened the pull request:
#### Describe the issue (if no issue has been made)
In OpenGLBuffer.cpp I noticed the following issues:
 - The vertex buffer reports it's size in bytes as the vertex count, and not count * sizeof(float)
 - ~~The constructor of OpenGLIndexBuffer used GL_ARRAY_BUFFER instead of GL_ELEMENT_ARRAY_BUFFER like it does in the same object's Bind() and Unbind() functions.~~

#### PR impact 
Issues this solves: closes #184 

#### Proposed fix
- [ ] Change OpenGLVertexBuffer to report the correct buffer size
- [ ] ~~Change OpenGLIndexBuffer to bind the correct buffer type in its constructor.~~ *Not bug

#### Additional context

LovelySanta commented:
The second bullet point is not a bug, as described in #95, and we have an open issue #184 to add a comment to it, to let people know this is how it should be.

MrLever commented:
> The second bullet point is not a bug, as described in #95, and we have an open issue #184 to add a comment to it, to let people know this is how it should be.

Reverted the change and added the comment mentioned in #184.

MrLever commented:
Oh, it seems that this was intentional. According to the video the argument to the VertexBuffer (size) is supposed to be the size in bytes.

LovelySanta commented:
Oh yes, I forgot (again) about that! As you see, the index buffer is taking an argument 'count', which is the number of elements, where that the vertex buffer is taking in the size (in bytes)! Can you edit it, we still can merge in the comment ;)

MrLever commented:
Fixed, only change is now the comment
